### PR TITLE
feat(cli): intro warning (30) support for RLS

### LINKED ISSUE #18700: Render warning code 30 in cli (PG RLS)
per https://github.com/prisma/prisma-engines/pull/3859

### PR DESCRIPTION
**Pull request by Druue**
closes #18700

### CODE REVIEW

File: packages/migrate/src/__tests__/fixtures/introspection/postgresql/row-level-security-warning-30/setup.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- Create a test table
+CREATE TABLE foo (
+    id SERIAL PRIMARY KEY,
+    -- We use this row to add security
```
Comment (janpio) [inline packages/migrate/src/__tests__/fixtures/introspection/postgresql/row-level-security-warning-30/setup.sql:4]: I think this comment is irrelevant here as we have no policy that uses the row.

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 2]
```diff
@@ -1685,17 +1685,103 @@ describe('postgresql partitioned tables introspection', () => {
...
+        'fixtures',
+        'introspection',
+        'postgresql',
+        `row-level-security-warning-${warningCode}${variant ? `-${variant}` : ''}`,
+      ),
+    }
+    return setupParams
+  }
+
+  const setupPostgressForWarning = (warningCode: number, variant?: number) => {
```
Comment (janpio) [inline packages/migrate/src/__tests__/DbPull.test.ts:1726]:
```suggestion
  const setupPostgresForWarning = (warningCode: number, variant?: number) => {
```
or is the `ss` intentional?

**Review by janpio — APPROVED** [review]

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 2 above)
Comment (Druue) [reply packages/migrate/src/__tests__/DbPull.test.ts:1726]: Ope, good catch, it was unintentional

File: packages/migrate/src/__tests__/DbPull.test.ts [hunk 3]
```diff
@@ -1685,17 +1685,103 @@ describe('postgresql partitioned tables introspection', () => {
...
+                                                                                                                                                                // - Model "measurement"
+                                                                                                                                                                // 
+                                                                                                                        `)
+      expect(ctx.mocked['process.stdout.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+      expect(ctx.mocked['process.stderr.write'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
+    })
+  })
+})
+
+describe('postgresql partitioned tables introspection', () => {
```
Comment (aqrln) [inline packages/migrate/src/__tests__/DbPull.test.ts:1705]:
```suggestion
describe('postgresql row level security introspection', () => {
```

File: packages/migrate/src/types.ts [hunk 4]
```diff
@@ -369,12 +370,17 @@ export namespace EngineArgs {
     affected: AffectedViewAndField[]
   }
 
-  // Partioned Tables
+  // Partitioned Tables
```
Comment (aqrln) [inline packages/migrate/src/types.ts:373]: Do we even need this comment? It doesn't seem to add much information here.

Comment (Druue) [reply packages/migrate/src/types.ts:373]: ope, it gets updated in another PR. when I had originally added partitioned tables, I figured there were going to be more warnings, but it's actually just going to be the one per stopgap. [Comparison](https://github.com/prisma/prisma/pull/18654/files#diff-2aeb8416b944a3e8efa6ac5e85341d7bfe5ac46b3557e7a083d43c2b5ab46321)

File: packages/migrate/src/__tests__/DbPull.test.ts (hunk 3 above)
Comment (Druue) [reply packages/migrate/src/__tests__/DbPull.test.ts:1705]: Ah, cheers!

**Review by aqrln — APPROVED** [review]

**Review by janpio — APPROVED** [review]
Code looks good.

**Review by janpio — APPROVED** [review]

File: packages/migrate/src/commands/DbPull.ts [hunk 5]
```diff
@@ -458,7 +458,8 @@ ${`Run ${chalk.green(getCommandWithExecutor('prisma generate'))} to generate Pri
           message += warning.affected.map((it) => `- Enum "${it.enm}", value: "${it.value}"`).join('\n')
         } else if (warning.code === 5 || warning.code === 6 || warning.code === 8) {
           message += warning.affected.map((it) => `- Model "${it.model}", field: "${it.field}"`).join('\n')
-        } else if (warning.code === 7 || warning.code === 14 || warning.code === 18 || warning.code === 19) {
+        } else if ([7, 14, 18, 19, 27, 30].includes(warning.code)) {
```
Comment (aqrln) [inline packages/migrate/src/commands/DbPull.ts:461]: This looks much nicer but makes the type of `warning.affected` below `any`, while explicit comparisons with each code allowed TS to narrow down the type.

Comment (Druue) [reply packages/migrate/src/commands/DbPull.ts:461]: Yeah, I'm trying to figure out a way to not lose the typing information 🤔

**Review by aqrln — APPROVED** [review]

**Pull request merged**